pydrofoil/test/test_supportcode.py

```diff
@@ -588,3 +591,513 @@ def test_softfloat_ui64tof64():
+        else:
+            assert res.tolong() == -((-frac.numerator) // frac.denominator)
+    
+@given(strategies.integers(), strategies.integers(min_value = 1))
+def test_real_floor_hypothesis(num, den):
+    r = rr_den_pos(num, den)
+    res = r.floor()
+    frac = Fraction(num, den)
+    if num % den == 0:
+        assert res.tolong() == frac.numerator
+    else:
+        if num > 0:
+            assert res.tolong() == frac.numerator // frac.denominator
+        else:
+            assert res.tolong() == -((-frac.numerator) // frac.denominator + 1)
+
+@given(strategies.integers(), strategies.integers(min_value = 1), strategies.integers(), strategies.integers(min_value = 1))
+def test_real_eq_hypothesis(num1, den1, num2, den2):
+    r1 = rr_den_pos(num1, den1)
+    r2 = rr_den_pos(num2, den2)
+    frac1 = Fraction(num1, den1)
+    frac2 = Fraction(num2, den2)
+    assert r1.eq(r2) == (frac1.numerator == frac2.numerator and frac1.denominator == frac2.denominator)
+
+@given(strategies.integers(), strategies.integers(min_value = 1), strategies.integers(), strategies.integers(min_value = 1))
+def test_real_lt_hypothesis(num1, den1, num2, den2):
+    r1 = rr_den_pos(num1, den1)
+    r2 = rr_den_pos(num2, den2)
+    frac1 = Fraction(num1, den1)
+    frac2 = Fraction(num2, den2)
+    assert r1.lt(r2) == (frac1 - frac2 < 0)
+
+@given(strategies.integers(), strategies.integers(min_value = 1), strategies.integers(), strategies.integers(min_value = 1))
+def test_real_gt_hypothesis(num1, den1, num2, den2):
+    r1 = rr_den_pos(num1, den1)
+    r2 = rr_den_pos(num2, den2)
+    frac1 = Fraction(num1, den1)
+    frac2 = Fraction(num2, den2)
+    assert r1.gt(r2) == (frac1 - frac2 > 0)
+
+@given(strategies.integers(), strategies.integers(min_value = 1), strategies.integers(), strategies.integers(min_value = 1))
+def test_real_le_hypothesis(num1, den1, num2, den2):
+    r1 = rr_den_pos(num1, den1)
+    r2 = rr_den_pos(num2, den2)
+    frac1 = Fraction(num1, den1)
+    frac2 = Fraction(num2, den2)
+    assert r1.le(r2) == (frac1 - frac2 <= 0)
+
+@given(strategies.integers(), strategies.integers(min_value = 1), strategies.integers(), strategies.integers(min_value = 1))
+def test_real_ge_hypothesis(num1, den1, num2, den2):
+    r1 = rr_den_pos(num1, den1)
+    r2 = rr_den_pos(num2, den2)
+    frac1 = Fraction(num1, den1)
+    frac2 = Fraction(num2, den2)
+    assert r1.ge(r2) == (frac1 - frac2 >= 0)
+
+@given(strategies.integers(), strategies.integers(min_value = 1), strategies.integers(min_value = 1, max_value = 50))
+def test_real_pow_hypothesis(num, den, n):
+    r = rr_den_pos(num, den)
+    res = r.pow(n)
+    frac = Fraction(num, den)
+    frac_pow = frac ** n
+    assert res.num.tolong() == frac_pow.numerator
+    assert res.den.tolong() == frac_pow.denominator
+
+@given(strategies.integers(), strategies.integers(min_value = 0))
+def test_real_fromstr_2_hypothesis(integer, fractional):
+    num_str = str(integer) + "." + str(fractional)
```

Comment:
the test data you generate has one limitation, it cannot generate this string: `"1.00001"`. you can fix it like this:

```python
@given(strategies.integers(), strategies.integers(min_value = 0, max_value=100), strategies.integers(min_value = 0))
def test_real_fromstr_2_hypothesis(integer, zeros, fractional):
    num_str = str(integer) + "." + "0" * zeros + str(fractional)
    ...
```

Reply:
ok, thanks for pointing out, I'm going to change it